emu/src/bitwise.rs

```diff
@@ -9,9 +9,8 @@ pub trait Bits {
     fn set_bit_off(&mut self, bit_idx: u8);
     fn toggle_bit(&mut self, bit_idx: u8);
     fn set_bit(&mut self, bit_idx: u8, value: bool);
-    fn get_bit(self, bit_idx: u8) -> bool;
-    fn get_bits(self, bits_range: RangeInclusive<u8>) -> u32;
-    fn are_bits_on(self, bits_range: RangeInclusive<u8>) -> bool;
```

Comment:
They are but I didn't remove them, I just changed from `self` to `&self` because to me it looked wrong

Reply:
Just because reviewing commit by commit was removed

Reply:
Yes I don't know why Github shows it in that way but you can see the same functions a few lines below with just `&` added

![image](https://user-images.githubusercontent.com/5256712/196240935-4047f3a6-853f-4e5f-b3f3-dbf7d3ab3ea2.png)

Reply:
> > > Can you squash latest 3 commit? They didn't add nothing relevant
> > 
> > 
> > You mean squashing the latest 3 commits into the [Memory: Add IORegister and LCDRegisters structs](https://github.com/RIP-Comm/clementine/pull/82/commits/cce2311fff2bce81ba60d19474de0cf3dd60fac6) one?
> 
> Yes, I mean if you have:
> 
> * ARM: Implement X instruction
> * ARM: Add test for X
> 
> Just keep the first commit, without other "fixes" or related, we don't have already PR merged then we can keep commit as clean as possibile :D

Ok I squashed 2 commits but I left the "Memory: Fix reading from work RAM" one since it changes something which was already present before this PR